### PR TITLE
Switch

### PR DESCRIPTION
**Pull request by mensfeld**
Was hanging few days now. This changes the API so please go through readme as well.

@filiptepper feel free to review it as well.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,16 @@
 # WaterDrop changelog
 
-## 0.4.0.1 - Unreleased
+## 1.0.0 unreleased
 
 - #37 - ack level for producer
 - Gem bump
 - Ruby 2.4.2 support
+- Raw ruby-kafka driver is not replaced with delivery_boy
```
Comment (nijikon) [inline CHANGELOG.md]: `not` -> `now`

File: README.md [hunk 2]
```diff
@@ -3,7 +3,15 @@
...
-Gem used to send messages to Kafka in an easy way.
+Gem used to send messages to Kafka in an easy way with an extra validation layer. It is a part of the [Karafka](https://github.com/karafka/karafka) ecosystem.
+
+WaterDrop is based on Zendesks [delivery_boy](https://github.com/zendesk/delivery_boy) gem.
+
+It is:
+
+ - Thread safe
+ - Supports sync producer
+ - Supports async producer
```
Comment (nijikon) [inline README.md]: We could changes this to `Supports sync and async producers`

File: README.md [hunk 3]
```diff
@@ -25,69 +33,79 @@ bundle install
...
 
 ```ruby
 WaterDrop.setup do |config|
-  config.send_messages = true
-  config.connection_pool.size = 20
-  config.connection_pool.timeout = 1
-  config.kafka.seed_brokers = ['localhost:9092']
-  config.raise_on_failure = true
+  config.deliver = true
+  config.kafka.seed_brokers = ['kafka://localhost:9092']
```
Comment (nijikon) [inline README.md]: `%w(kafka://localhost:9092)`

File: README.md [hunk 4]
```diff
@@ -100,18 +118,18 @@ message that you want to send should be either binary or stringified (to_s, to_j
...
-Each pull request must pass our quality requirements. To check if everything is as it should be, we use [Coditsu](https://coditsu.io) that combines multiple linters and code analyzers for both code and documentation.
+Each pull request must pass our quality requirements. To check if everything is as it should be, we use [Coditsu](https://coditsu.io) that combinse multiple linters and code analyzers for both code and documentation.
 
 Unfortunately, it does not yet support independent forks, however you should be fine by looking at what we require.
 
 Please run:
 
 ```bash
-bundle exec rake
+bundle exec rspec spec
```
Comment (nijikon) [inline README.md]: No need to pass `spec`, `bundle exec rspec` is enough

File: README.md [hunk 5]
```diff
@@ -100,18 +118,18 @@ message that you want to send should be either binary or stringified (to_s, to_j
 Fork the project.
 Make your feature addition or bug fix.
 Add tests for it. This is important so we don't break it in a future versions unintentionally.
-Commit, do not mess with Rakefile, version, or history. (if you want to have your own version, that is fine but bump version in a commit by itself I can ignore when I pull). Send me a pull request. Bonus points for topic branches.
+Commit, do not mess with version, or history. (if you want to have your own version, that is fine but bump version in a commit by itself I can ignore when I pull). Send me a pull request. Bonus points for topic branches.
 
 [![coditsu](https://coditsu.io/assets/quality_bar.svg)](https://app.coditsu.io/karafka/repositories/waterdrop)
 
-Each pull request must pass our quality requirements. To check if everything is as it should be, we use [Coditsu](https://coditsu.io) that combines multiple linters and code analyzers for both code and documentation.
+Each pull request must pass our quality requirements. To check if everything is as it should be, we use [Coditsu](https://coditsu.io) that combinse multiple linters and code analyzers for both code and documentation.
```
Comment (nijikon) [inline README.md]: `combinse` -> `combines`

File: lib/water_drop/sync_producer.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+# WaterDrop library
+module WaterDrop
+  # Sync producer for messages
+  SyncProducer = Class.new(BaseProducer)
+  # Sync producer for messages
+  Producer = SyncProducer
```
Comment (nijikon) [inline lib/water_drop/sync_producer.rb]: This is backwards compatibility?

Comment (mensfeld) [reply lib/water_drop/sync_producer.rb]: No. It is just an alias for convenience. 1.0 is totally not compatible with 0.4

Comment (nijikon) [reply lib/water_drop/sync_producer.rb]: I would remove this, we should be strict, either `SyncProducer` or `AsyncProducer` imho.

**Pull request merged**